src/pages/ProfilePage.js

```diff
@@ -158,6 +158,30 @@ const Dashboard = () => {
     ],
     ["Referral Bonus Per Transaction", `$ ${referralRate?.fixedAmount}`]
   ];
+
+  const onConnectXero = async () => {
+    try {
+      const url = await gpib.secure.get("/xero");
+      if (url) {
+        window.open(url.data, "_blank");
+      }
+    } catch (error) {
+      console.error(error);
+      setErrorMessage(error);
+    }
+  };
+
+  const syncBankAccount = async () => {
+    setSyncingBankAccount(true);
+    try {
+      await gpib.secure.post("/xero/syncBankAccount");
+    } catch (error) {
+      console.error(error);
+      setErrorMessage(error);
+      setSyncingBankAccount(false);
+    }
+    setSyncingBankAccount(false);
```

Comment:
I'm likely wrong but wouldn't this be run twice if it runs the catch?
```suggestion
    }
    setSyncingBankAccount(false);
```

Reply:
@Mitchell139 you're correct.